backend.py

```diff
@@ -440,6 +441,22 @@ def performInstallation(answers, ui_package, interactive):
                                "latest dedicated ISO." %
                                (answers['linstor-version'], ', '.join(available_linstor_versions)))
 
+    # if upgrading a host with xapi, check we can upgrade it first
+    if answers['install-type'] == INSTALL_TYPE_REINSTALL and answers['xapi-version']:
+        available_xapi_versions = repository.listPackagesFromRepos(
+            main_repositories, 'xapi-core', '%{evr}')
+        if not available_xapi_versions:
+            raise RuntimeError("No xapi package found")
+        available_xapi_version = available_xapi_versions[0]
+        for v in available_xapi_versions[1:]:
```

Comment:
This `[1:]` leads me to think some comments would be useful here.

Note that `listPackagesFromRepos` is written with the assumption that repoquery will return all available versions (as `dnf repoquery` does) but quick tests right now show only a single version of each package in CentOS7 (like `--latest-limit=1` with `dnf`) - that makes me even more puzzled about that slice.

Reply:
that's just to select the greatest available version, in case there would be more than one returned by repoquery.

Reply:
Oh I only now realize that's a "latest version in list" here.
IMO we'll rather want to make sure `repoquery` does that on its own, and ensure it returns a single `xapi-core` version.

The fact `listPackagesFromRepos` only returns a single now today is a bug (assuming the CentOS7 `repoquery` was acting more like `dnf repoquery`).
I'd go with a flag for to return only the latest, add `--show-duplicates` for when it isn't set, and an `assert len()==1`  when set to make we adjust flags properly when we'll switch to `dnf` for v9.

Reply:
The linstor case would use `latest_only=False`.  But today we only support upgrading it from an ISO, on which the only available version will be the latest one.